tests/test_models.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""Test module."""
+
+from __future__ import absolute_import, print_function
+
+import pytest
+from invenio_files_rest.models import Bucket, ObjectVersion
+from sqlalchemy.orm.exc import NoResultFound
+
+from invenio_records_files.api import Record
+from invenio_records_files.models import RecordsBuckets
+
+
+def test_cascade_action_record_delete_force_false(app, db, location, record,
+                                                  generic_file):
+    """Test cascade action on record delete, with force false."""
+    record_id = record.id
+    bucket_id = record.files.bucket.id
+
+    # check before
+    assert len(RecordsBuckets.query.all()) == 1
+    assert len(Bucket.query.all()) == 1
+    assert len(Bucket.query.filter_by(id=bucket_id).all()) == 1
+    assert ObjectVersion.get(bucket=bucket_id, key=generic_file)
+
+    record.delete(force=False)
```

Comment:
this can be done as parameterized test:

```python
@pytest.parameterize('force,number_of_buckets', [(False, 1), (True, 0)])
def def test_cascade_action_record_delete(..., force, number_of_buckets):
    ...
    record.delete(force=force)
    ...
    assert len(RecordsBuckets.query.all()) == number_of_buckets
```